components/calendar/src/iso.rs

```diff
@@ -458,7 +466,7 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1237-L1258
     pub(crate) fn iso_from_fixed(date: i32) -> Date<Iso> {
         let year = Self::iso_year_from_fixed(date);
-        let prior_days = date - Self::iso_new_year(year);
+        let prior_days = date.saturating_sub(Self::iso_new_year(year));
```

Comment:
I can think of two solutions here:

1. Change the logic to look at Jan 1 of the following year instead of Jan 1 of the current year. This is more involved and requires diverging from the book.
2. Special-case very negative years to shift the date into the following year for the purposes of month and day-of-month calculations. First figure out how many days are in the year, and then add that to `date`. Note that you need to be very careful around leap days.